src/main/haxe/yloader/impl/js/NodeLoader.hx

```diff
@@ -0,0 +1,187 @@
+			case "https:":
+				return createHttpsRequest(url);
+			default:
+				return null;
+		}
+	}
+
+	function addClientRequestListeners():Void
+	{
+		clientRequest.on("error", onClientRequestError);
+	}
+
+	function removeClientRequestListeners():Void
+	{
+		clientRequest.removeListener("error", onClientRequestError);
+	}
+
+	function writeData():Void
+	{
+		if (requestHasData())
+			clientRequest.write(request.data);
+	}
+
+	function endRequest():Void
+	{
+		clientRequest.end();
+	}
+
+	function createHttpRequest(url:UrlData):ClientRequest
+	{
+		var options:HttpRequestOptions = createOptions(url);
+		return Http.request(options, onRequestResponse);
+	}
+
+	function createHttpsRequest(url:UrlData):ClientRequest
+	{
+		var options:HttpsRequestOptions = createOptions(url);
+		return Https.request(options, onRequestResponse);
+	}
+
+	function createOptions(url:UrlData):HttpsRequestOptions
+	{
+		return {
+			host: url.host,
+			port: Std.parseInt(url.port),
+			path: url.path,
+			method: Std.string(request.method),
+			headers: getHeaders()
+		};
+	}
+
+	function getHeaders():Dynamic
+	{
+		var result:Dynamic = {};
+		var isContentLengthPresent:Bool = false;
+
+		for (header in request.headers)
+		{
+			Reflect.setField(result, header.name, header.value);
+			if (header.name.toLowerCase() == HeaderUtil.CONTENT_LENGTH_HEADER_NAME)
+				isContentLengthPresent = true;
+		}
+		if (!isContentLengthPresent && requestHasData())
+			Reflect.setField(result, HeaderUtil.CONTENT_LENGTH_HEADER_NAME, Std.string(Buffer.byteLength(request.data)));
+
+		return result;
+	}
+
+	function requestHasData():Bool
+	{
+		return request.data != null && request.data.length > 0;
+	}
+
+	function handleResponse(responseObject:Dynamic)
+	{
+		if(onResponse == null || response == null)
+			return;
+
+		var response = getResponse(responseObject);
+		onResponse(response);
+	}
+
+	function getResponse(responseObject:Dynamic):Response
+	{
+		return new Response(isSuccess(response.statusCode), responseObject, response.statusCode,
+			response.statusMessage, fromRawHeadersToArray(response.rawHeaders));
+	}
+
+	function isSuccess(statusCode:Int):Bool
+	{
+		return StatusCodeUtil.isSuccess(statusCode);
+	}
+
+	function fromRawHeadersToArray(source:Array<String>):Array<Parameter>
+	{
+		var i = 0, result:Array<Parameter> = [];
+		while (i < source.length)
+			result.push(new Parameter(source[i++], source[i++]));
+
+		return result;
+	}
+
+	function onRequestResponse(response:IncomingMessage):Void
+	{
+		this.response = response;
+		response.setEncoding(responseEncoding);
+		response.on("data", onIncomingMessageData);
+		response.on("end", onIncomingMessageEnd);
+	}
+
+	function onIncomingMessageData(data:String):Void
+	{
+		responseString += data;
+	}
+
+	function onIncomingMessageEnd(data:Dynamic):Void
+	{
+		handleResponse(responseString);
+		cancel();
+	}
+
+	function onClientRequestError(error:Dynamic):Void
+	{
+		onResponse(new Response(false, null, 0, error.message, null));
```

Comment:
missing onResponse null check

Reply:
is cancel needed here?

Reply:
I would also remove error.message, as its Dynamic it can be anything and any type